src/photon/photon_scheduler.c

```diff
@@ -53,6 +56,184 @@ void print_resource_info(const local_scheduler_state *state,
 #endif
 }
 
+/**
+ * Kill a worker, if it is a child process, and clean up all of its associated
+ * state.
+ *
+ * @param worker A pointer to the worker we want to kill.
+ * @param wait A bool representing whether we should wait for the worker's
+ *        process to exit. If the worker is not a child process, this flag is
+ *        ignored.
+ * @return Void.
+ */
+void kill_worker(local_scheduler_client *worker, bool wait) {
```

Comment:
When we kill a worker, we probably need to let the scheduling algorithm do some clean up as well (in particular, the scheduling algorithm needs to remove the worker struct from its list of available workers, otherwise it will try to assign a task to that worker).